lms/djangoapps/certificates/models.py

```diff
@@ -304,20 +304,19 @@ def certificate_info_for_user(user, course_id, grade, user_is_whitelisted=None):
             user=user, course_id=course_id, whitelist=True
         ).exists()
 
+    user_is_eligible = 'N'
+    certificate_is_delivered = 'N'
+    certificate_type = 'N/A'
     eligible_for_certificate = (user_is_whitelisted or grade is not None) and user.profile.allow_certificate
 
     if eligible_for_certificate:
```

Comment:
Can get rid of this additional variable.
user_is_eligible = 'Y' if (user_is_whitelisted or grade is not None) and user.profile.allow_certificate else 'N'